### DISCUSSION TITLE
Screen Colour Filter

### DISCUSSION THREAD

**Discussion by LoveBodhi** [Wohlstand/TheXTech#680] [Ideas]
- Sepia
- Black and White
- Inverted
- Retro-style
- ... (Or maybe custom)

**Comment by ds-sloth** [comment]
It's a great idea to discuss this! Do you think that screen filters should be a per-section setting, or a global setting?

**Reply by ChristianSilvermoon** [reply]
Per Section would let people do a lot of interesting things, but global could also be really nice... could this be considered a Shaders thing?

**Reply by ds-sloth** [reply]
Definitely. I also think that global and per-section have very different use cases. Things like sepia or inverted seem quite likely to be used per-section. Things like the NDS color filter or hq2x filter you mentioned in #338 seem more likely to be useful globally.

Most of the logic for per-section was already implemented for compatibility with the X2 file format, but global would require an extra user setting and some extra logic to hook it up.

Asset packs will be able to package their own per-section screen filters (and also global screen filters), but I'm very interested to see a list of the per-section effects people want to be included with the main asset packs. We could probably start by reimplementing most of the X2 screen filters.

**Reply by LoveBodhi** [reply]
This may be implmented into NsPr Deluxe Edition.
